### PR TITLE
New: Added responsive support for image sizes in Media control [ED-2108]

### PR DESCRIPTION
**Pull request by KingYes**

### CODE REVIEW

File: assets/dev/js/editor/controls/media.js [hunk 1]
```diff
@@ -65,10 +66,20 @@ ControlMediaItemView = ControlMultipleBaseItemView.extend( {
 		}
 
 		if ( this.ui.mediaInputImageSize ) {
-			this.ui.mediaInputImageSize.val( this.getControlValue( 'size' ) );
+			let imageSize = this.getControlValue( 'size' );
+
+			if ( isPlaceholder ) {
+				imageSize = this.getControlPlaceholder()?.size;
```
Comment (StyleShit) [inline assets/dev/js/editor/controls/media.js:72]:
```suggestion
				imageSize = this.getControlPlaceholder()?.size ?? imageSIze;
```

Maybe?

File: assets/dev/js/editor/controls/media.js [hunk 2]
```diff
@@ -121,12 +132,38 @@ ControlMediaItemView = ControlMultipleBaseItemView.extend( {
 			return;
 		}
 
-		const currentControlValue = this.getControlValue();
+		const currentControlValue = this.getControlValue(),
+			placeholder = this.getControlPlaceholder();
+
+		const isSelectedImage = ( '' !== currentControlValue?.id ),
+			isSelectedPlaceholder = placeholder?.id;
```
Comment (StyleShit) [inline assets/dev/js/editor/controls/media.js:139]:
```suggestion
		const hasImage = ( '' !== currentControlValue?.id ),
			hasPlaceholder = placeholder?.id;
```

File: assets/dev/js/editor/controls/media.js [hunk 3]
```diff
@@ -121,12 +132,38 @@ ControlMediaItemView = ControlMultipleBaseItemView.extend( {
...
+
+		if ( ! isSelectedImage && ! isSelectedPlaceholder ) {
```
Comment (StyleShit) [inline assets/dev/js/editor/controls/media.js:141]: Shouldn't be `||`?

File: assets/dev/js/editor/controls/media.js [hunk 4]
```diff
@@ -121,12 +132,38 @@ ControlMediaItemView = ControlMultipleBaseItemView.extend( {
...
+			if ( this.model.get( 'responsive' ) ) {
+				// Render is already calls `applySavedValue`, therefore there's no need for it in this case.
+				this.renderWithChildren();
+			} else {
+				this.applySavedValue();
+			}
+
+			this.onMediaInputImageSizeChange();
+
+			return;
```
Comment (StyleShit) [inline assets/dev/js/editor/controls/media.js:160]: maybe extract to function?

File: assets/dev/js/editor/controls/media.js [hunk 5]
```diff
@@ -121,12 +132,38 @@ ControlMediaItemView = ControlMultipleBaseItemView.extend( {
...
+		}
+
+		if ( ! isSelectedImage && isSelectedPlaceholder ) {
+			this.setValue( {
+				...placeholder,
+				size: currentControlValue.size,
+			} );
+
+			if ( this.model.get( 'responsive' ) ) {
+				// Render is already calls `applySavedValue`, therefore there's no need for it in this case.
```
Comment (StyleShit) [inline assets/dev/js/editor/controls/media.js:152]:
```suggestion
				// Render already calls `applySavedValue`, therefore there's no need for it in this case.
```

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/controls/media.js [hunk 6]
```diff
@@ -121,12 +132,38 @@ ControlMediaItemView = ControlMultipleBaseItemView.extend( {
...
+			placeholder = this.getControlPlaceholder();
+
+		const isSelectedImage = ( '' !== currentControlValue?.id ),
+			isSelectedPlaceholder = placeholder?.id;
+
+		if ( ! isSelectedImage && ! isSelectedPlaceholder ) {
+			return;
+		}
+
+		if ( ! isSelectedImage && isSelectedPlaceholder ) {
```
Comment (StyleShit) [inline assets/dev/js/editor/controls/media.js:145]: Can you extract the condition to a meaningful variable?

File: assets/dev/js/editor/controls/media.js (hunk 1 above)
Comment (KingYes) [reply assets/dev/js/editor/controls/media.js:72]: Nope

File: assets/dev/js/editor/controls/media.js (hunk 4 above)
Comment (KingYes) [reply assets/dev/js/editor/controls/media.js:160]: Try to it

File: assets/dev/js/editor/controls/media.js (hunk 5 above)
Comment (KingYes) [reply assets/dev/js/editor/controls/media.js:152]: It's copy-paste from other code

File: assets/dev/js/editor/controls/media.js (hunk 6 above)
Comment (KingYes) [reply assets/dev/js/editor/controls/media.js:145]: Like what?

**Pull request merged**